app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -58,6 +106,20 @@ const SurveyAnimals: React.FC = () => {
     }
   };
 
+  const handleTelemetrySave = async (survey_critter_id: number, data: IAnimalTelemetryDevice) => {
+    const critter = critterData?.find((a) => a.survey_critter_id === survey_critter_id);
+    (data as any).critter_id = critter?.critter_id;
+    console.log('Data to be sent ' + JSON.stringify(data));
+    await bhApi.survey.addDeployment(
+      projectId,
+      surveyId,
+      survey_critter_id,
+      data as IAnimalTelemetryDevice & { critter_id: string }
```

Comment:
Could we not just change the `data` function parameter to be type `IAnimalTelemetryDevice & { critter_id: string }`? This would allow us to avoid casting